test/Microsoft.AspNetCore.SignalR.Client.Tests/ConnectionTests.cs

```diff
@@ -437,6 +437,118 @@ public void CannotStartConnectionWithInvalidTransportType(TransportType requeste
         }
 
         [Fact]
+        public async Task EventQueueTimeout()
+        {
+            var mockHttpHandler = new Mock<HttpMessageHandler>();
+            mockHttpHandler.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .Returns<HttpRequestMessage, CancellationToken>(async (request, cancellationToken) =>
+                {
+                    await Task.Yield();
+                    return request.Method == HttpMethod.Options
+                        ? ResponseUtils.CreateResponse(HttpStatusCode.OK, ResponseUtils.CreateNegotiationResponse())
+                        : ResponseUtils.CreateResponse(HttpStatusCode.OK);
+                });
+
+            var mockTransport = new Mock<ITransport>();
+            Channel<byte[], SendMessage> channel = null;
+            mockTransport.Setup(t => t.StartAsync(It.IsAny<Uri>(), It.IsAny<Channel<byte[], SendMessage>>()))
+                .Returns<Uri, Channel<byte[], SendMessage>>((url, c) =>
+                {
+                    channel = c;
+                    return Task.CompletedTask;
+                });
+            mockTransport.Setup(t => t.StopAsync())
+                .Returns(() =>
+                {
+                    channel.Out.TryComplete();
+                    return Task.CompletedTask;
+                });
+
+
+            var callbackInvokedTcs = new TaskCompletionSource<object>();
+            var closedTcs = new TaskCompletionSource<object>();
+
+            var connection = new HttpConnection(new Uri("http://fakeuri.org/"), new TestTransportFactory(mockTransport.Object), loggerFactory: null, httpMessageHandler: mockHttpHandler.Object);
+            connection.Received +=
+                async m =>
+                {
+                    await Task.Delay(6000);
+                    callbackInvokedTcs.SetResult(null);
+                    await closedTcs.Task;
+                };
+
+            await connection.StartAsync();
+            channel.Out.TryWrite(Array.Empty<byte>());
+
+            // Ensure that the Received callback has been called before attempting the second write
```

Comment:
I don't think you should send twice. (Does it mean the test takes now 12 seconds?)